checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -575,8 +591,27 @@ private void computeFieldAccessType(AnnotatedTypeMirror type,
                     .asType();
             boolean isInitializedForFrame = isInitializedForFrame(receiverType, fieldDeclarationType);
             if (isInitializedForFrame) {
+                // receiver is initialized for this frame.
+                // change the type of the field to @UnknownInitialization so that
+                // anything can be assigned to this field.
                 type.replaceAnnotation(qualHierarchy.getTopAnnotation(UNCLASSIFIED));
+
+            } else if(computingAnnotatedTypeMirrorOfLHS) {
+                // receiver is not initialized for this frame and the type of a lhs is being computed
+                // so replace all annotations with top expect for the field invariant annotation hierarchy
+                // (Nullness, for example)
+                AnnotationMirror childTypeSystemAnnotation = type.getAnnotationInHierarchy(getFieldInvariantAnnotation());
```

Comment:
The documentation was misleading. By "field invariant annotation hierarchy" I meant "child hierarchy".  I refactored this code to check whether an annotation is in an initialization hierarchy rather than checking for annotations in the child hierarchy.